store/src/db/mod.rs

```diff
@@ -347,8 +346,14 @@ impl Db {
                         let transaction = conn.transaction()?;
                         let accounts: Vec<_> = account_smt
                             .leaves()
-                            .map(|(account_id, state_hash)| (account_id, None, state_hash.into()))
-                            .collect();
+                            .map(|(account_id, state_hash)| {
+                                Ok(AccountDetailsUpdate {
+                                    account_id: account_id.try_into()?,
+                                    final_state_hash: state_hash.into(),
+                                    details: None,
+                                })
```

Comment:
Does this assume that there are no public accounts in the genesis file? If so, we should probably create an issue about handling public accounts in the genesis state.

Reply:
Yes, we don't support public accounts in the genesis fill, will create an issue about it, thank you!

Reply:
Created an issue: https://github.com/0xPolygonMiden/miden-node/issues/305